### PR TITLE
helper funcs for converting rootcfgs to terraform roots

### PR DESCRIPTION
**Pull request by smonero**
- Note that I don't think we need things like `TriggerInfo` and plan mode so we don't set them.
- Also I just use the string representations from the deploy package rather that copy/pasting them (EnvStep, Manifest, etc)
- We could also use a WorkflowSignaler with TemporalClient nil to use the helper funcs like prependPlanEnvSteps but I figured it would be more straightforward to just copy them
- We can also potentially move things to an `internal` folder similar to what is done with the pr and deploy flows, this would require changing some of the imports but not sure if that is even useful
- Originally I wanted to reuse the structs from the deploy and pr flows (i.e. root, repo, step, etc), I decided to go with the `deploy` version rather than the `pr` version because `deploy` has the Apply job. However I am not 100% sure if that is actually necessary.

relies on https://github.com/lyft/atlantis/pull/732 and https://github.com/lyft/atlantis/pull/743 to be merged first, then we will be able to actually use these funcs.

### CODE REVIEW

File: server/neptune/workflows/adhoc.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+/*
+type PRStep = request.Step
+type PRPlanMode = request.PlanMode
+type PRAppCredentials = request.AppCredentials
+type PRRequest = pr.Request
+
+func PR(ctx workflow.Context, request PRRequest) error {
+	return pr.Workflow(ctx, request, Terraform)
+}
+*/
```
Comment (samrabelachew) [inline server/neptune/workflows/adhoc.go:21]: nit: remove

File: server/neptune/adhoc/adhocexecutionhelpers/adhoc_execution_params.go [hunk 2]
```diff
@@ -16,15 +19,29 @@ type AdhocTerraformWorkflowExecutionParams struct {
...
+func ConstructAdhocExecParamsWithRootCfgBuilderAndRepoRetriever(ctx context.Context, repoName string, revision string, githubRetriever *adhocgithubhelpers.AdhocGithubRetriever, rootCfgBuilder *root_config.Builder) (AdhocTerraformWorkflowExecutionParams, error) {
 	// TODO: in the future, could potentially pass in the owner instead of hardcoding lyft
 	repo, err := githubRetriever.GetRepository(ctx, "lyft", repoName)
 	if err != nil {
 		return AdhocTerraformWorkflowExecutionParams{}, errors.Wrap(err, "getting repo")
 	}
 
+	opts := config.BuilderOptions{
+		RepoFetcherOptions: &internal_gh.RepoFetcherOptions{
+			CloneDepth: 1,
```
Comment (samrabelachew) [inline server/neptune/adhoc/adhocexecutionhelpers/adhoc_execution_params.go:31]: It's not required, but do you want to enable SimplePath here?

**Review by samrabelachew — COMMENTED** [review]
Can you write tests for these changes?

File: server/neptune/adhoc/adhocexecutionhelpers/adhoc_execution_params.go (hunk 2 above)
Comment (smonero) [reply server/neptune/adhoc/adhocexecutionhelpers/adhoc_execution_params.go:31]: oh yeah I forgot

**Review by samrabelachew — APPROVED** [review]

**Pull request merged**